<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Summarizing Data Examples | Data Analysis and Numerical Methods for Biological Systems Engineers</title>
  <meta name="description" content="<p>This book was developed as a companion text for the Virginia Tech Biological
Systems Engineering course BSE3144: Engineering Analysis for Biological
Systems using Numerical Methods.</p>" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Summarizing Data Examples | Data Analysis and Numerical Methods for Biological Systems Engineers" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This book was developed as a companion text for the Virginia Tech Biological
Systems Engineering course BSE3144: Engineering Analysis for Biological
Systems using Numerical Methods.</p>" />
  <meta name="github-repo" content="VT-BSE3144/bookdown" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Summarizing Data Examples | Data Analysis and Numerical Methods for Biological Systems Engineers" />
  
  <meta name="twitter:description" content="<p>This book was developed as a companion text for the Virginia Tech Biological
Systems Engineering course BSE3144: Engineering Analysis for Biological
Systems using Numerical Methods.</p>" />
  

<meta name="author" content="Durelle Scott" />
<meta name="author" content="R. Clay Wright" />


<meta name="date" content="2025-04-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="summarizing-data.html"/>
<link rel="next" href="data-visualization.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Analysis and Numerical Methods</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About this book</a></li>
<li class="chapter" data-level="1" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html"><i class="fa fa-check"></i><b>1</b> Introduction to Posit, Rstudio, &amp; Markdown</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#pre-work"><i class="fa fa-check"></i><b>1.1</b> Pre-work</a></li>
<li class="chapter" data-level="1.2" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#posit"><i class="fa fa-check"></i><b>1.2</b> Posit</a></li>
<li class="chapter" data-level="1.3" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#rstudio"><i class="fa fa-check"></i><b>1.3</b> Rstudio</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-source-pane"><i class="fa fa-check"></i><b>1.3.1</b> The Source pane</a></li>
<li class="chapter" data-level="1.3.2" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#markdown"><i class="fa fa-check"></i><b>1.3.2</b> Markdown</a></li>
<li class="chapter" data-level="1.3.3" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#more-markdown-formatting"><i class="fa fa-check"></i><b>1.3.3</b> More Markdown formatting</a></li>
<li class="chapter" data-level="1.3.4" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#running-code"><i class="fa fa-check"></i><b>1.3.4</b> Running code</a></li>
<li class="chapter" data-level="1.3.5" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-console-pane"><i class="fa fa-check"></i><b>1.3.5</b> The Console pane</a></li>
<li class="chapter" data-level="1.3.6" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-help-pane"><i class="fa fa-check"></i><b>1.3.6</b> The Help pane</a></li>
<li class="chapter" data-level="1.3.7" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-filesplotspackageshelpviewer-pane"><i class="fa fa-check"></i><b>1.3.7</b> The Files/Plots/Packages/Help/Viewer pane</a></li>
<li class="chapter" data-level="1.3.8" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#ok-back-to-the-source-pane"><i class="fa fa-check"></i><b>1.3.8</b> OK, back to the Source pane…</a></li>
<li class="chapter" data-level="1.3.9" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-environment-pane"><i class="fa fa-check"></i><b>1.3.9</b> The Environment pane</a></li>
<li class="chapter" data-level="1.3.10" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#the-historyconnectionsgitbuildtutorial-pane"><i class="fa fa-check"></i><b>1.3.10</b> The History/Connections/Git/Build/Tutorial pane</a></li>
<li class="chapter" data-level="1.3.11" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#terminalrenderbackground-jobs"><i class="fa fa-check"></i><b>1.3.11</b> Terminal/Render/Background Jobs</a></li>
<li class="chapter" data-level="1.3.12" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#yaml-headers"><i class="fa fa-check"></i><b>1.3.12</b> YAML headers</a></li>
<li class="chapter" data-level="1.3.13" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#code-chunkblock-settings-and-global-settings"><i class="fa fa-check"></i><b>1.3.13</b> Code chunk/block settings and global settings</a></li>
<li class="chapter" data-level="1.3.14" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#output-options"><i class="fa fa-check"></i><b>1.3.14</b> Output Options</a></li>
<li class="chapter" data-level="1.3.15" data-path="introduction-to-posit-rstudio-markdown.html"><a href="introduction-to-posit-rstudio-markdown.html#troubleshooting-knitrender-problems"><i class="fa fa-check"></i><b>1.3.15</b> Troubleshooting Knit/Render problems</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html"><i class="fa fa-check"></i>Rstudio and Markdown examples</a>
<ul>
<li class="chapter" data-level="1.4" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#navigate-the-posit-and-rstudio-interface-with-proficiency."><i class="fa fa-check"></i><b>1.4</b> Navigate the Posit and RStudio interface with proficiency.</a>
<ul>
<li class="chapter" data-level="1.4.1" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#making-new-documents"><i class="fa fa-check"></i><b>1.4.1</b> Making new documents</a></li>
<li class="chapter" data-level="1.4.2" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#navigating-the-file-pane-and-file-paths"><i class="fa fa-check"></i><b>1.4.2</b> Navigating the file pane and file paths</a></li>
<li class="chapter" data-level="1.4.3" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#console-pane"><i class="fa fa-check"></i><b>1.4.3</b> Console pane</a></li>
<li class="chapter" data-level="1.4.4" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#environment-pane"><i class="fa fa-check"></i><b>1.4.4</b> Environment pane</a></li>
<li class="chapter" data-level="1.4.5" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#history-pane"><i class="fa fa-check"></i><b>1.4.5</b> History pane</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#articulate-the-process-of-using-rstudio-to-develop-markdown-documents."><i class="fa fa-check"></i><b>1.5</b> Articulate the process of using RStudio to develop markdown documents.</a>
<ul>
<li class="chapter" data-level="1.5.1" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#cheat-sheets-yay"><i class="fa fa-check"></i><b>1.5.1</b> Cheat Sheets! Yay!</a></li>
<li class="chapter" data-level="1.5.2" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#inserting-code-chunks"><i class="fa fa-check"></i><b>1.5.2</b> Inserting code chunks</a></li>
<li class="chapter" data-level="1.5.3" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#changing-code-chunk-settings"><i class="fa fa-check"></i><b>1.5.3</b> Changing code chunk settings</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#practice-using-markdown-to-format-reproducible-documents"><i class="fa fa-check"></i><b>1.6</b> Practice using markdown to format reproducible documents</a>
<ul>
<li class="chapter" data-level="1.6.1" data-path="rstudio-and-markdown-examples.html"><a href="rstudio-and-markdown-examples.html#tasks"><i class="fa fa-check"></i><b>1.6.1</b> Tasks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html"><i class="fa fa-check"></i><b>2</b> The R Language and Tidy Data</a>
<ul>
<li class="chapter" data-level="2.0.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#data-organization-and-manipulation-in-spreadsheets"><i class="fa fa-check"></i><b>2.0.1</b> Data Organization and manipulation in Spreadsheets</a></li>
<li class="chapter" data-level="2.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#reading-and-pre-class-materials"><i class="fa fa-check"></i><b>2.1</b> Reading and Pre-class Materials</a></li>
<li class="chapter" data-level="2.2" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#tidy-data"><i class="fa fa-check"></i><b>2.2</b> Tidy data</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#top-12-tips-for-data-organization"><i class="fa fa-check"></i><b>2.2.1</b> Top 12 Tips for Data Organization</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#r-basics"><i class="fa fa-check"></i><b>2.3</b> R Basics</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#basic-math-to-linear-algebra"><i class="fa fa-check"></i><b>2.3.1</b> Basic Math to Linear Algebra</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#functions"><i class="fa fa-check"></i><b>2.4</b> Functions</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#what-is-a-function"><i class="fa fa-check"></i><b>2.4.1</b> What is a function?</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#objects---variables-and-functions"><i class="fa fa-check"></i><b>2.5</b> Objects - Variables and Functions</a>
<ul>
<li class="chapter" data-level="2.5.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#obey-rs-naming-rules"><i class="fa fa-check"></i><b>2.5.1</b> Obey R’s naming rules</a></li>
<li class="chapter" data-level="2.5.2" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#variables-and-data-types"><i class="fa fa-check"></i><b>2.5.2</b> Variables and data types</a></li>
<li class="chapter" data-level="2.5.3" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#vectors"><i class="fa fa-check"></i><b>2.5.3</b> Vectors</a></li>
<li class="chapter" data-level="2.5.4" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#matrices-arrays-and-lists"><i class="fa fa-check"></i><b>2.5.4</b> Matrices, Arrays and Lists</a></li>
<li class="chapter" data-level="2.5.5" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#data-frames"><i class="fa fa-check"></i><b>2.5.5</b> Data frames</a></li>
<li class="chapter" data-level="2.5.6" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#run-a-function"><i class="fa fa-check"></i><b>2.5.6</b> Run a function</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#tidyverse-tables"><i class="fa fa-check"></i><b>2.6</b> Tidyverse tables</a>
<ul>
<li class="chapter" data-level="2.6.1" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#create-a-table-manually"><i class="fa fa-check"></i><b>2.6.1</b> Create a Table Manually</a></li>
<li class="chapter" data-level="2.6.2" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#tribble"><i class="fa fa-check"></i><b>2.6.2</b> <code>tribble()</code></a></li>
<li class="chapter" data-level="2.6.3" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#read-a-csv-file-.csv"><i class="fa fa-check"></i><b>2.6.3</b> Read a CSV file (.csv)</a></li>
<li class="chapter" data-level="2.6.4" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#read-a-character-delimited-file-.txt"><i class="fa fa-check"></i><b>2.6.4</b> Read a character-delimited file (.txt)</a></li>
<li class="chapter" data-level="2.6.5" data-path="the-r-language-and-tidy-data.html"><a href="the-r-language-and-tidy-data.html#read-an-excel-file-.xls-.xlsx"><i class="fa fa-check"></i><b>2.6.5</b> Read an Excel file (.xls, .xlsx)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html"><i class="fa fa-check"></i>The R Language and Tidy Data Examples</a>
<ul>
<li class="chapter" data-level="2.7" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#tidy-data-1"><i class="fa fa-check"></i><b>2.7</b> Tidy Data</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#lets-make-a-data-dictionary-for-this-dataset"><i class="fa fa-check"></i><b>2.7.1</b> Let’s make a data dictionary for this dataset</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#r-basics-1"><i class="fa fa-check"></i><b>2.8</b> R Basics</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#data-types"><i class="fa fa-check"></i><b>2.8.1</b> Data types</a></li>
<li class="chapter" data-level="2.8.2" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#functions-1"><i class="fa fa-check"></i><b>2.8.2</b> Functions</a></li>
<li class="chapter" data-level="2.8.3" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#objects"><i class="fa fa-check"></i><b>2.8.3</b> Objects</a></li>
<li class="chapter" data-level="2.8.4" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#vectors-1"><i class="fa fa-check"></i><b>2.8.4</b> Vectors</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#data-tibbles-dataframes"><i class="fa fa-check"></i><b>2.9</b> Data, tibbles, dataframes</a>
<ul>
<li class="chapter" data-level="2.9.1" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#reading-in-data"><i class="fa fa-check"></i><b>2.9.1</b> Reading in data</a></li>
<li class="chapter" data-level="2.9.2" data-path="the-r-language-and-tidy-data-examples.html"><a href="the-r-language-and-tidy-data-examples.html#factors-1"><i class="fa fa-check"></i><b>2.9.2</b> Factors</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html"><i class="fa fa-check"></i><b>3</b> Functions, Best-coding Practices, and Debugging Intro</a>
<ul>
<li class="chapter" data-level="3.1" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#functions-2"><i class="fa fa-check"></i><b>3.1</b> Functions</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#functions-why-and-how"><i class="fa fa-check"></i><b>3.1.1</b> Functions… why and how</a></li>
<li class="chapter" data-level="3.1.2" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#a-simple-function-cv"><i class="fa fa-check"></i><b>3.1.2</b> A simple function: CV()</a></li>
<li class="chapter" data-level="3.1.3" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#now-we-need-to-apply-our-function."><i class="fa fa-check"></i><b>3.1.3</b> Now we need to apply our function.</a></li>
<li class="chapter" data-level="3.1.4" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#what-objects-are-available"><i class="fa fa-check"></i><b>3.1.4</b> What objects are available?</a></li>
<li class="chapter" data-level="3.1.5" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#optional-arguments"><i class="fa fa-check"></i><b>3.1.5</b> Optional arguments</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#packages"><i class="fa fa-check"></i><b>3.2</b> Packages</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#installing-and-using-packages"><i class="fa fa-check"></i><b>3.2.1</b> Installing and using packages</a></li>
<li class="chapter" data-level="3.2.2" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#installing-and-using-packages-1"><i class="fa fa-check"></i><b>3.2.2</b> Installing and using packages</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#best-coding-practices"><i class="fa fa-check"></i><b>3.3</b> Best coding practices</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#reproducibility"><i class="fa fa-check"></i><b>3.3.1</b> Reproducibility</a></li>
<li class="chapter" data-level="3.3.2" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#style-guides"><i class="fa fa-check"></i><b>3.3.2</b> Style guides</a></li>
<li class="chapter" data-level="3.3.3" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#automated-code-stylingformatting"><i class="fa fa-check"></i><b>3.3.3</b> Automated code styling/formatting</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#debugging"><i class="fa fa-check"></i><b>3.4</b> Debugging</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#verifying-code"><i class="fa fa-check"></i><b>3.4.1</b> Verifying code</a></li>
<li class="chapter" data-level="3.4.2" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#commenting-your-code"><i class="fa fa-check"></i><b>3.4.2</b> Commenting your code</a></li>
<li class="chapter" data-level="3.4.3" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#commenting-out-code"><i class="fa fa-check"></i><b>3.4.3</b> “Commenting out” code</a></li>
<li class="chapter" data-level="3.4.4" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#common-errors-and-how-to-fix-them"><i class="fa fa-check"></i><b>3.4.4</b> Common errors and How to Fix Them</a></li>
<li class="chapter" data-level="3.4.5" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#warning-or-error"><i class="fa fa-check"></i><b>3.4.5</b> Warning or Error?</a></li>
<li class="chapter" data-level="3.4.6" data-path="functions-best-coding-practices-and-debugging-intro.html"><a href="functions-best-coding-practices-and-debugging-intro.html#debugging-is-scientific"><i class="fa fa-check"></i><b>3.4.6</b> Debugging is scientific</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html"><i class="fa fa-check"></i>Functions, Best-coding Practices, and Debugging Examples</a>
<ul>
<li class="chapter" data-level="3.5" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#functions-3"><i class="fa fa-check"></i><b>3.5</b> Functions</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#example-1-1"><i class="fa fa-check"></i><b>3.5.1</b> Example 1</a></li>
<li class="chapter" data-level="3.5.2" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#example-2-1"><i class="fa fa-check"></i><b>3.5.2</b> Example 2</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#best-coding-practices-1"><i class="fa fa-check"></i><b>3.6</b> Best coding practices</a>
<ul>
<li class="chapter" data-level="3.6.1" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#code-from-inside-out-running-the-smallest-bits-of-code-possible"><i class="fa fa-check"></i><b>3.6.1</b> Code from inside out, running the smallest bits of code possible</a></li>
<li class="chapter" data-level="3.6.2" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#use-rstudio-to-your-advantage"><i class="fa fa-check"></i><b>3.6.2</b> Use RStudio to your advantage</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#debugging-1"><i class="fa fa-check"></i><b>3.7</b> Debugging</a>
<ul>
<li class="chapter" data-level="3.7.1" data-path="functions-best-coding-practices-and-debugging-examples.html"><a href="functions-best-coding-practices-and-debugging-examples.html#also-use-rstudio"><i class="fa fa-check"></i><b>3.7.1</b> Also use RStudio</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>4</b> Data Wrangling</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-wrangling.html"><a href="data-wrangling.html#connection-to-previous-work-on-data-organization"><i class="fa fa-check"></i><b>4.1</b> Connection to previous work on Data Organization</a></li>
<li class="chapter" data-level="4.2" data-path="data-wrangling.html"><a href="data-wrangling.html#source"><i class="fa fa-check"></i><b>4.2</b> Source</a></li>
<li class="chapter" data-level="4.3" data-path="data-wrangling.html"><a href="data-wrangling.html#introduction---example-dataset-nycflights13"><i class="fa fa-check"></i><b>4.3</b> 3.1 Introduction - Example dataset <code>nycflights13</code></a></li>
<li class="chapter" data-level="4.4" data-path="data-wrangling.html"><a href="data-wrangling.html#section"><i class="fa fa-check"></i><b>4.4</b> 3.1.2</a></li>
<li class="chapter" data-level="4.5" data-path="data-wrangling.html"><a href="data-wrangling.html#dplyr-basics"><i class="fa fa-check"></i><b>4.5</b> 3.1.3 dplyr basics</a></li>
<li class="chapter" data-level="4.6" data-path="data-wrangling.html"><a href="data-wrangling.html#row-wise-functions"><i class="fa fa-check"></i><b>4.6</b> 3.2 Row-wise functions</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="data-wrangling.html"><a href="data-wrangling.html#filter"><i class="fa fa-check"></i><b>4.6.1</b> 3.2.1 filter()</a></li>
<li class="chapter" data-level="4.6.2" data-path="data-wrangling.html"><a href="data-wrangling.html#arranging-rows"><i class="fa fa-check"></i><b>4.6.2</b> 3.2.3 Arranging rows</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="data-wrangling.html"><a href="data-wrangling.html#columns"><i class="fa fa-check"></i><b>4.7</b> 3.3 Columns</a>
<ul>
<li class="chapter" data-level="4.7.1" data-path="data-wrangling.html"><a href="data-wrangling.html#mutate---add-new-variables"><i class="fa fa-check"></i><b>4.7.1</b> 3.3.1 Mutate - Add new variables</a></li>
<li class="chapter" data-level="4.7.2" data-path="data-wrangling.html"><a href="data-wrangling.html#select"><i class="fa fa-check"></i><b>4.7.2</b> 3.3.2 Select</a></li>
</ul></li>
<li class="chapter" data-level="4.8" data-path="data-wrangling.html"><a href="data-wrangling.html#the-pipe"><i class="fa fa-check"></i><b>4.8</b> 3.4 The pipe</a></li>
<li class="chapter" data-level="4.9" data-path="data-wrangling.html"><a href="data-wrangling.html#groups"><i class="fa fa-check"></i><b>4.9</b> 3.5 Groups</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html"><i class="fa fa-check"></i>Data Wrangling Examples</a>
<ul>
<li class="chapter" data-level="4.9.1" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html#filter-1"><i class="fa fa-check"></i><b>4.9.1</b> Filter</a></li>
<li class="chapter" data-level="4.9.2" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html#arrange"><i class="fa fa-check"></i><b>4.9.2</b> Arrange</a></li>
<li class="chapter" data-level="4.9.3" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html#mutate---adds-a-new-column"><i class="fa fa-check"></i><b>4.9.3</b> Mutate - adds a new column</a></li>
<li class="chapter" data-level="4.9.4" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html#select-1"><i class="fa fa-check"></i><b>4.9.4</b> Select</a></li>
<li class="chapter" data-level="4.9.5" data-path="data-wrangling-examples.html"><a href="data-wrangling-examples.html#grouped-summaries"><i class="fa fa-check"></i><b>4.9.5</b> Grouped summaries</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="summarizing-data.html"><a href="summarizing-data.html"><i class="fa fa-check"></i><b>5</b> Summarizing Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="summarizing-data.html"><a href="summarizing-data.html#learning-objectives"><i class="fa fa-check"></i><b>5.1</b> Learning Objectives</a></li>
<li class="chapter" data-level="5.2" data-path="summarizing-data.html"><a href="summarizing-data.html#introduction"><i class="fa fa-check"></i><b>5.2</b> Introduction</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="summarizing-data.html"><a href="summarizing-data.html#readings-complete-by-class-on-monday"><i class="fa fa-check"></i><b>5.2.1</b> Readings (complete by class on Monday)</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="summarizing-data.html"><a href="summarizing-data.html#group_by-function-summarizing-info-based-on-typecharacteristic"><i class="fa fa-check"></i><b>5.3</b> <code>group_by</code> function: Summarizing info based on type/characteristic</a></li>
<li class="chapter" data-level="5.4" data-path="summarizing-data.html"><a href="summarizing-data.html#grouping-with-multiple-variables"><i class="fa fa-check"></i><b>5.4</b> Grouping with multiple variables</a></li>
<li class="chapter" data-level="5.5" data-path="summarizing-data.html"><a href="summarizing-data.html#tidying-and-untidying-data-with-pivot_-functions"><i class="fa fa-check"></i><b>5.5</b> Tidying and untidying data with <code>pivot_</code> functions</a></li>
<li class="chapter" data-level="5.6" data-path="summarizing-data.html"><a href="summarizing-data.html#joins-and-binds-within-dplyr"><i class="fa fa-check"></i><b>5.6</b> Joins and Binds within <code>dplyr</code></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html"><i class="fa fa-check"></i>Summarizing Data Examples</a>
<ul>
<li class="chapter" data-level="5.7" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#bioreactor-data-analysis"><i class="fa fa-check"></i><b>5.7</b> Bioreactor Data Analysis</a>
<ul>
<li class="chapter" data-level="5.7.1" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#background"><i class="fa fa-check"></i><b>5.7.1</b> Background</a></li>
<li class="chapter" data-level="5.7.2" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#tasks-1"><i class="fa fa-check"></i><b>5.7.2</b> Tasks</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#root-growth-inhibition-example"><i class="fa fa-check"></i><b>5.8</b> Root Growth Inhibition Example</a>
<ul>
<li class="chapter" data-level="5.8.1" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#introduction-1"><i class="fa fa-check"></i><b>5.8.1</b> Introduction</a></li>
<li class="chapter" data-level="5.8.2" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#read-in-data"><i class="fa fa-check"></i><b>5.8.2</b> Read in data</a></li>
<li class="chapter" data-level="5.8.3" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#write-a-function-for-tidying-this-data"><i class="fa fa-check"></i><b>5.8.3</b> Write a function for tidying this data</a></li>
<li class="chapter" data-level="5.8.4" data-path="summarizing-data-examples.html"><a href="summarizing-data-examples.html#tidy-the-individual-datasets"><i class="fa fa-check"></i><b>5.8.4</b> Tidy the individual datasets</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="data-visualization.html"><a href="data-visualization.html"><i class="fa fa-check"></i><b>6</b> Data Visualization</a>
<ul>
<li class="chapter" data-level="6.1" data-path="data-visualization.html"><a href="data-visualization.html#reading-complete-by-class-on-monday"><i class="fa fa-check"></i><b>6.1</b> Reading (complete by class on Monday)</a></li>
<li class="chapter" data-level="6.2" data-path="data-visualization.html"><a href="data-visualization.html#tutorials-to-walk-through-on-your-own"><i class="fa fa-check"></i><b>6.2</b> Tutorials to walk through on your own</a></li>
<li class="chapter" data-level="6.3" data-path="data-visualization.html"><a href="data-visualization.html#more-on-color-palettes"><i class="fa fa-check"></i><b>6.3</b> More on Color Palettes</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="data-visualization.html"><a href="data-visualization.html#previewing"><i class="fa fa-check"></i><b>6.3.1</b> Previewing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-visualization-examples.html"><a href="data-visualization-examples.html"><i class="fa fa-check"></i>Data Visualization Examples</a>
<ul>
<li class="chapter" data-level="6.4" data-path="data-visualization-examples.html"><a href="data-visualization-examples.html#background-1"><i class="fa fa-check"></i><b>6.4</b> Background</a></li>
<li class="chapter" data-level="6.5" data-path="data-visualization-examples.html"><a href="data-visualization-examples.html#data-description"><i class="fa fa-check"></i><b>6.5</b> Data Description</a></li>
<li class="chapter" data-level="6.6" data-path="data-visualization-examples.html"><a href="data-visualization-examples.html#tasks-2"><i class="fa fa-check"></i><b>6.6</b> Tasks</a></li>
<li class="chapter" data-level="6.7" data-path="data-visualization-examples.html"><a href="data-visualization-examples.html#setup-for-the-homework"><i class="fa fa-check"></i><b>6.7</b> Setup for the Homework</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="assessment-1.html"><a href="assessment-1.html"><i class="fa fa-check"></i><b>7</b> Assessment 1</a>
<ul>
<li class="chapter" data-level="7.1" data-path="assessment-1.html"><a href="assessment-1.html#directions"><i class="fa fa-check"></i><b>7.1</b> Directions</a></li>
<li class="chapter" data-level="7.2" data-path="assessment-1.html"><a href="assessment-1.html#finding-a-data-set"><i class="fa fa-check"></i><b>7.2</b> Finding a data set</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="assessment-1.html"><a href="assessment-1.html#general"><i class="fa fa-check"></i><b>7.2.1</b> General</a></li>
<li class="chapter" data-level="7.2.2" data-path="assessment-1.html"><a href="assessment-1.html#health"><i class="fa fa-check"></i><b>7.2.2</b> Health</a></li>
<li class="chapter" data-level="7.2.3" data-path="assessment-1.html"><a href="assessment-1.html#environment"><i class="fa fa-check"></i><b>7.2.3</b> Environment</a></li>
<li class="chapter" data-level="7.2.4" data-path="assessment-1.html"><a href="assessment-1.html#agriculture"><i class="fa fa-check"></i><b>7.2.4</b> Agriculture</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="assessment-1.html"><a href="assessment-1.html#questionstasks"><i class="fa fa-check"></i><b>7.3</b> Questions/Tasks</a></li>
<li class="chapter" data-level="7.4" data-path="assessment-1.html"><a href="assessment-1.html#guideline-on-the-use-of-ai-tools"><i class="fa fa-check"></i><b>7.4</b> Guideline on the Use of AI Tools</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html"><i class="fa fa-check"></i><b>8</b> Introduction to Linear Regression in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#objectives"><i class="fa fa-check"></i><b>8.1</b> Objectives</a></li>
<li class="chapter" data-level="8.2" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#reading"><i class="fa fa-check"></i><b>8.2</b> Reading</a></li>
<li class="chapter" data-level="8.3" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#active-package-libraries"><i class="fa fa-check"></i><b>8.3</b> Active Package Libraries</a></li>
<li class="chapter" data-level="8.4" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#linear-regression"><i class="fa fa-check"></i><b>8.4</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="8.4.1" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#assumptions"><i class="fa fa-check"></i><b>8.4.1</b> Assumptions</a></li>
<li class="chapter" data-level="8.4.2" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#workflow"><i class="fa fa-check"></i><b>8.4.2</b> Workflow:</a></li>
<li class="chapter" data-level="8.4.3" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#example-8"><i class="fa fa-check"></i><b>8.4.3</b> Example</a></li>
<li class="chapter" data-level="8.4.4" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#predicting-values-applications-of-linear-models"><i class="fa fa-check"></i><b>8.4.4</b> Predicting values: Applications of linear models</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#now-lets-say-you-want-to-predict-a-y-value-for-a-given-age."><i class="fa fa-check"></i><b>8.5</b> Now, let’s say you want to predict a y-value for a given age.</a></li>
<li class="chapter" data-level="8.6" data-path="introduction-to-linear-regression-in-r.html"><a href="introduction-to-linear-regression-in-r.html#example-2-from-reading"><i class="fa fa-check"></i><b>8.6</b> Example 2 from reading</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html"><i class="fa fa-check"></i>Linear Regression Examples</a>
<ul>
<li class="chapter" data-level="8.7" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#active-libraries"><i class="fa fa-check"></i><b>8.7</b> Active Libraries</a></li>
<li class="chapter" data-level="8.8" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#linear-regression-1"><i class="fa fa-check"></i><b>8.8</b> Linear Regression</a></li>
<li class="chapter" data-level="8.9" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#assumptions-1"><i class="fa fa-check"></i><b>8.9</b> Assumptions</a></li>
<li class="chapter" data-level="8.10" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#example-10"><i class="fa fa-check"></i><b>8.10</b> Example</a>
<ul>
<li class="chapter" data-level="8.10.1" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-1.-read-in-data-1"><i class="fa fa-check"></i><b>8.10.1</b> Step 1. Read in data</a></li>
<li class="chapter" data-level="8.10.2" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-2.-plot-data-visualize-linearity-1"><i class="fa fa-check"></i><b>8.10.2</b> Step 2. Plot data &amp; visualize linearity</a></li>
<li class="chapter" data-level="8.10.3" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-3.-transform-data-1"><i class="fa fa-check"></i><b>8.10.3</b> Step 3. Transform data</a></li>
<li class="chapter" data-level="8.10.4" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-4.-linear-model-1"><i class="fa fa-check"></i><b>8.10.4</b> Step 4. Linear model</a></li>
<li class="chapter" data-level="8.10.5" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-5.-test-for-homoscedasaticity-1"><i class="fa fa-check"></i><b>8.10.5</b> Step 5. Test for homoscedasaticity</a></li>
<li class="chapter" data-level="8.10.6" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-6.-test-for-bias-1"><i class="fa fa-check"></i><b>8.10.6</b> Step 6. Test for bias</a></li>
<li class="chapter" data-level="8.10.7" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-7.-test-for-normality-1"><i class="fa fa-check"></i><b>8.10.7</b> Step 7. Test for Normality</a></li>
<li class="chapter" data-level="8.10.8" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#application"><i class="fa fa-check"></i><b>8.10.8</b> Application</a></li>
</ul></li>
<li class="chapter" data-level="8.11" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#take-2-try-second-order"><i class="fa fa-check"></i><b>8.11</b> Take 2, try second order!</a>
<ul>
<li class="chapter" data-level="8.11.1" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-4.-linear-model-2"><i class="fa fa-check"></i><b>8.11.1</b> Step 4. Linear model</a></li>
<li class="chapter" data-level="8.11.2" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-5.-test-for-homoscedasaticity-2"><i class="fa fa-check"></i><b>8.11.2</b> Step 5. Test for homoscedasaticity</a></li>
<li class="chapter" data-level="8.11.3" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-6.-test-for-bias-2"><i class="fa fa-check"></i><b>8.11.3</b> Step 6. Test for bias</a></li>
<li class="chapter" data-level="8.11.4" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#step-7.-test-for-normality-2"><i class="fa fa-check"></i><b>8.11.4</b> Step 7. Test for Normality</a></li>
<li class="chapter" data-level="8.11.5" data-path="linear-regression-examples.html"><a href="linear-regression-examples.html#application-1"><i class="fa fa-check"></i><b>8.11.5</b> Application</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html"><i class="fa fa-check"></i><b>9</b> Intro to ANOVA and non-linear least squares regression</a>
<ul>
<li class="chapter" data-level="9.1" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#reading-1"><i class="fa fa-check"></i><b>9.1</b> Reading</a></li>
<li class="chapter" data-level="9.2" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#anova-tests-interactions-between-variables"><i class="fa fa-check"></i><b>9.2</b> ANOVA tests interactions between variables</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#experiment-1"><i class="fa fa-check"></i><b>9.2.1</b> Experiment 1</a></li>
<li class="chapter" data-level="9.2.2" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#ploting-the-full-dataset"><i class="fa fa-check"></i><b>9.2.2</b> Ploting the full dataset</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#post-hoc-testing"><i class="fa fa-check"></i><b>9.3</b> Post hoc testing</a></li>
<li class="chapter" data-level="9.4" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#assumptions-of-anova"><i class="fa fa-check"></i><b>9.4</b> Assumptions of ANOVA</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#normality-of-residuals"><i class="fa fa-check"></i><b>9.4.1</b> Normality of residuals</a></li>
<li class="chapter" data-level="9.4.2" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#homogeneity-of-variance"><i class="fa fa-check"></i><b>9.4.2</b> Homogeneity of variance</a></li>
<li class="chapter" data-level="9.4.3" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#independence"><i class="fa fa-check"></i><b>9.4.3</b> Independence</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="intro-to-anova-and-non-linear-least-squares-regression.html"><a href="intro-to-anova-and-non-linear-least-squares-regression.html#non-linear-least-squares"><i class="fa fa-check"></i><b>9.5</b> Non-linear least squares</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="anova-and-non-linear-least-squares-regression-examples.html"><a href="anova-and-non-linear-least-squares-regression-examples.html"><i class="fa fa-check"></i>ANOVA and non-linear least squares regression examples</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="anova-and-non-linear-least-squares-regression-examples.html"><a href="anova-and-non-linear-least-squares-regression-examples.html#lineweaver-burk-plot"><i class="fa fa-check"></i><b>9.5.1</b> Lineweaver-Burk plot</a></li>
<li class="chapter" data-level="9.6" data-path="anova-and-non-linear-least-squares-regression-examples.html"><a href="anova-and-non-linear-least-squares-regression-examples.html#nonlinear-least-squares-regression"><i class="fa fa-check"></i><b>9.6</b> Nonlinear least squares regression</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Data Analysis and Numerical Methods for Biological Systems Engineers</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="summarizing-data-examples" class="section level1 unnumbered hasAnchor">
<h1>Summarizing Data Examples<a href="summarizing-data-examples.html#summarizing-data-examples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="summarizing-data-examples.html#cb331-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## ── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
## ✔ dplyr     1.1.4     ✔ readr     2.1.5
## ✔ forcats   1.0.0     ✔ stringr   1.5.1
## ✔ ggplot2   3.5.1     ✔ tibble    3.2.1
## ✔ lubridate 1.9.4     ✔ tidyr     1.3.1
## ✔ purrr     1.0.4     
## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()
## ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="summarizing-data-examples.html#cb333-1" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code></pre></div>
<div id="bioreactor-data-analysis" class="section level2 hasAnchor" number="5.7">
<h2><span class="header-section-number">5.7</span> Bioreactor Data Analysis<a href="summarizing-data-examples.html#bioreactor-data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="background" class="section level3 hasAnchor" number="5.7.1">
<h3><span class="header-section-number">5.7.1</span> Background<a href="summarizing-data-examples.html#background" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In a bioprocess engineering lab, several bioreactor runs were performed to study the effect of different feed types on process performance. Two datasets were collected:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Bioreactor Readings</strong><br />
This dataset (<code>bioreactor_readings.csv</code>) contains time-series measurements recorded during each run. It is provided in a wide format with columns for the run number, time (in hours), and various sensor readings:</p>
<ul>
<li><p><code>Run</code>: Identifier for the bioreactor run.</p></li>
<li><p><code>Time</code>: Time stamp (hours).</p></li>
<li><p><code>Temp</code>: Temperature (°C).</p></li>
<li><p><code>pH</code>: pH level.</p></li>
<li><p><code>DO</code>: Dissolved Oxygen (mg/L).</p></li>
<li><p><code>Substrate</code>: Substrate concentration (g/L).</p></li>
</ul></li>
<li><p><strong>Experimental Metadata</strong><br />
This dataset (<code>experiment_info.csv</code>) contains metadata about each run:</p>
<ul>
<li><p><code>Run</code>: Run identifier (matching the bioreactor dataset).</p></li>
<li><p><code>Feed</code>: Type of feed (e.g., “Glucose”, “Glycerol”).</p></li>
<li><p><code>Inoculum</code>: Inoculum concentration (OD units).</p></li>
<li><p><code>Operator</code>: Name of the operator in charge.</p></li>
</ul></li>
</ol>
</div>
<div id="tasks-1" class="section level3 hasAnchor" number="5.7.2">
<h3><span class="header-section-number">5.7.2</span> Tasks<a href="summarizing-data-examples.html#tasks-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="import-and-inspect-the-data" class="section level4 hasAnchor" number="5.7.2.1">
<h4><span class="header-section-number">5.7.2.1</span> 1. Import and Inspect the Data<a href="summarizing-data-examples.html#import-and-inspect-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>-   Load both CSV files into R as data frames.

-   Use the `head()` function (or similar) to inspect the first few rows of each dataset.</code></pre>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="summarizing-data-examples.html#cb335-1" tabindex="-1"></a><span class="co"># Load libraries</span></span>
<span id="cb335-2"><a href="summarizing-data-examples.html#cb335-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb335-3"><a href="summarizing-data-examples.html#cb335-3" tabindex="-1"></a></span>
<span id="cb335-4"><a href="summarizing-data-examples.html#cb335-4" tabindex="-1"></a><span class="co"># Import data</span></span>
<span id="cb335-5"><a href="summarizing-data-examples.html#cb335-5" tabindex="-1"></a>bioreactor <span class="ot">&lt;-</span></span>
<span id="cb335-6"><a href="summarizing-data-examples.html#cb335-6" tabindex="-1"></a>  <span class="fu">read_csv</span>(<span class="st">&quot;data/bioreactor_readings.csv&quot;</span>)</span></code></pre></div>
<pre><code>## Rows: 10 Columns: 6
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;,&quot;
## dbl (6): Run, Time, Temp, pH, DO, Substrate
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="summarizing-data-examples.html#cb337-1" tabindex="-1"></a>experiment <span class="ot">&lt;-</span></span>
<span id="cb337-2"><a href="summarizing-data-examples.html#cb337-2" tabindex="-1"></a>  <span class="fu">read_csv</span>(<span class="st">&quot;data/experiment_info.csv&quot;</span>)</span></code></pre></div>
<pre><code>## Rows: 2 Columns: 4
## ── Column specification ────────────────────────────────────────────────────────
## Delimiter: &quot;,&quot;
## chr (2): Feed, Operator
## dbl (2): Run, Inoculum
## 
## ℹ Use `spec()` to retrieve the full column specification for this data.
## ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="summarizing-data-examples.html#cb339-1" tabindex="-1"></a><span class="fu">head</span>(bioreactor)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##     Run  Time  Temp    pH    DO Substrate
##   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;
## 1     1     0  30     7     6.5       1.5
## 2     1     1  30.5   7.1   6.4       1.3
## 3     1     2  31     7.2   6.3       1.1
## 4     1     3  31.2   7.1   6.2       0.9
## 5     1     4  31     7     6         0.8
## 6     2     0  29.5   6.8   6.8       1</code></pre>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="summarizing-data-examples.html#cb341-1" tabindex="-1"></a><span class="fu">head</span>(experiment)</span></code></pre></div>
<pre><code>## # A tibble: 2 × 4
##     Run Feed     Inoculum Operator
##   &lt;dbl&gt; &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;   
## 1     1 Glucose      0.15 Alice   
## 2     2 Glycerol     0.12 Bob</code></pre>
</div>
<div id="data-tidying-pivoting" class="section level4 hasAnchor" number="5.7.2.2">
<h4><span class="header-section-number">5.7.2.2</span> 2. Data Tidying (Pivoting)<a href="summarizing-data-examples.html#data-tidying-pivoting" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>-   **Pivot Longer:**\
    Convert the bioreactor readings from wide format into a long (tidy) format so that all measurements (Temp, pH, DO, Substrate) are in a single column named `Parameter` with corresponding values in a column named `Value`.\
    *Hint: Use `tidyr::pivot_longer()`.*

-   **Pivot Wider:**\
    After some analysis, you decide to reshape the long dataset back into a wide format, but now grouping by `Run` and `Time` such that each parameter becomes its own column again.\
    *Hint: Use `tidyr::pivot_wider()`.*
    </code></pre>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="summarizing-data-examples.html#cb344-1" tabindex="-1"></a><span class="co"># 2. Pivot longer: Tidy the bioreactor data</span></span>
<span id="cb344-2"><a href="summarizing-data-examples.html#cb344-2" tabindex="-1"></a>bioreactor_long <span class="ot">&lt;-</span> bioreactor <span class="sc">%&gt;%</span></span>
<span id="cb344-3"><a href="summarizing-data-examples.html#cb344-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> Temp<span class="sc">:</span>Substrate, </span>
<span id="cb344-4"><a href="summarizing-data-examples.html#cb344-4" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;Parameter&quot;</span>, </span>
<span id="cb344-5"><a href="summarizing-data-examples.html#cb344-5" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;Value&quot;</span>)</span>
<span id="cb344-6"><a href="summarizing-data-examples.html#cb344-6" tabindex="-1"></a></span>
<span id="cb344-7"><a href="summarizing-data-examples.html#cb344-7" tabindex="-1"></a><span class="co"># This would allow us to plot both parameters on the same plot</span></span>
<span id="cb344-8"><a href="summarizing-data-examples.html#cb344-8" tabindex="-1"></a></span>
<span id="cb344-9"><a href="summarizing-data-examples.html#cb344-9" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> bioreactor_long, </span>
<span id="cb344-10"><a href="summarizing-data-examples.html#cb344-10" tabindex="-1"></a>       <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> Time, </span>
<span id="cb344-11"><a href="summarizing-data-examples.html#cb344-11" tabindex="-1"></a>                     <span class="at">y =</span> Value, </span>
<span id="cb344-12"><a href="summarizing-data-examples.html#cb344-12" tabindex="-1"></a>                     <span class="at">color =</span> Parameter, </span>
<span id="cb344-13"><a href="summarizing-data-examples.html#cb344-13" tabindex="-1"></a>                     <span class="at">linetype =</span> <span class="fu">as.factor</span>(Run))) <span class="sc">+</span> </span>
<span id="cb344-14"><a href="summarizing-data-examples.html#cb344-14" tabindex="-1"></a>  <span class="fu">geom_line</span>()</span></code></pre></div>
<p><img src="05-1_Examples_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
</div>
<div id="b.-pivot-wider-reshape-back-to-wide-format-if-needed" class="section level4 hasAnchor" number="5.7.2.3">
<h4><span class="header-section-number">5.7.2.3</span> 2b. Pivot wider: Reshape back to wide format if needed<a href="summarizing-data-examples.html#b.-pivot-wider-reshape-back-to-wide-format-if-needed" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="summarizing-data-examples.html#cb345-1" tabindex="-1"></a>bioreactor_wide <span class="ot">&lt;-</span> bioreactor_long <span class="sc">%&gt;%</span></span>
<span id="cb345-2"><a href="summarizing-data-examples.html#cb345-2" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Parameter, <span class="at">values_from =</span> Value)</span></code></pre></div>
</div>
<div id="joining-datasets" class="section level4 hasAnchor" number="5.7.2.4">
<h4><span class="header-section-number">5.7.2.4</span> 3. Joining Datasets<a href="summarizing-data-examples.html#joining-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>-   Merge the reshaped bioreactor dataset with the experimental metadata using the common `Run` column.\
    *Hint: Use one of the join functions (e.g., `dplyr::left_join()`).*
    </code></pre>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="summarizing-data-examples.html#cb347-1" tabindex="-1"></a><span class="co"># 3. Join datasets by &#39;Run&#39;</span></span>
<span id="cb347-2"><a href="summarizing-data-examples.html#cb347-2" tabindex="-1"></a>merged_data <span class="ot">&lt;-</span> bioreactor_wide <span class="sc">%&gt;%</span></span>
<span id="cb347-3"><a href="summarizing-data-examples.html#cb347-3" tabindex="-1"></a>  <span class="fu">left_join</span>(experiment, <span class="at">by =</span> <span class="st">&quot;Run&quot;</span>)</span></code></pre></div>
</div>
<div id="grouping-and-summarizing" class="section level4 hasAnchor" number="5.7.2.5">
<h4><span class="header-section-number">5.7.2.5</span> 4. Grouping and Summarizing<a href="summarizing-data-examples.html#grouping-and-summarizing" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>-   Group the merged dataset by the `Feed` type.

-   Calculate summary statistics (e.g., mean and standard deviation, or min and max) for key measurements such as `Temp`, `pH`, and `Substrate` across all time points and runs for each feed type.\
    *Hint: Use `dplyr::group_by()` and `dplyr::summarize()`.*</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="summarizing-data-examples.html#cb349-1" tabindex="-1"></a><span class="co"># 4. Group and summarize by &#39;Feed&#39;</span></span>
<span id="cb349-2"><a href="summarizing-data-examples.html#cb349-2" tabindex="-1"></a>summary_stats <span class="ot">&lt;-</span> merged_data <span class="sc">%&gt;%</span></span>
<span id="cb349-3"><a href="summarizing-data-examples.html#cb349-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Feed) <span class="sc">%&gt;%</span></span>
<span id="cb349-4"><a href="summarizing-data-examples.html#cb349-4" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">mean_Temp =</span> <span class="fu">mean</span>(Temp, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb349-5"><a href="summarizing-data-examples.html#cb349-5" tabindex="-1"></a>            <span class="at">sd_Temp   =</span> <span class="fu">sd</span>(Temp, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb349-6"><a href="summarizing-data-examples.html#cb349-6" tabindex="-1"></a>            <span class="at">mean_pH   =</span> <span class="fu">mean</span>(pH, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb349-7"><a href="summarizing-data-examples.html#cb349-7" tabindex="-1"></a>            <span class="at">sd_pH     =</span> <span class="fu">sd</span>(pH, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb349-8"><a href="summarizing-data-examples.html#cb349-8" tabindex="-1"></a>            <span class="at">max_Subs  =</span> <span class="fu">max</span>(Substrate, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), </span>
<span id="cb349-9"><a href="summarizing-data-examples.html#cb349-9" tabindex="-1"></a>            <span class="at">min_Subs  =</span> <span class="fu">min</span>(Substrate, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb349-10"><a href="summarizing-data-examples.html#cb349-10" tabindex="-1"></a></span>
<span id="cb349-11"><a href="summarizing-data-examples.html#cb349-11" tabindex="-1"></a><span class="co"># View summary statistics</span></span>
<span id="cb349-12"><a href="summarizing-data-examples.html#cb349-12" tabindex="-1"></a><span class="fu">print</span>(summary_stats)</span></code></pre></div>
<pre><code>## # A tibble: 2 × 7
##   Feed     mean_Temp sd_Temp mean_pH  sd_pH max_Subs min_Subs
##   &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
## 1 Glucose       30.7   0.488    7.08 0.0837      1.5      0.8
## 2 Glycerol      30.2   0.488    6.88 0.0837      1.1      1</code></pre>
<ol start="5" style="list-style-type: decimal">
<li><p><strong>Interpreting the Results</strong></p>
<ul>
<li><p>Discuss how different feed types glucose vs glycerol may be affecting the reactor conditions based on your summary statistics.</p></li>
<li><p>What potential insights could an engineer draw from these analyses?</p></li>
</ul></li>
</ol>
</div>
</div>
</div>
<div id="root-growth-inhibition-example" class="section level2 hasAnchor" number="5.8">
<h2><span class="header-section-number">5.8</span> Root Growth Inhibition Example<a href="summarizing-data-examples.html#root-growth-inhibition-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction-1" class="section level3 hasAnchor" number="5.8.1">
<h3><span class="header-section-number">5.8.1</span> Introduction<a href="summarizing-data-examples.html#introduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In my research group we study plant genes and how they function. One interesting thing about plants is that their genomes are quite large compared to other organisms and one of the reasons for that is their genomes have been duplicated several times throughout the history of their evolution</p>
<div class="float">
<img src="data/Largest-collection-of-garden-gnomes-60at60-photo2_tcm25-386214.webp" alt="gnome duplication" />
<div class="figcaption">gnome duplication</div>
</div>
<p>These genome duplications have generated families of genes which have similar but slightly different functions. Often we think about each of these genes in a family as playing a distinct role, and by <strong>adding</strong> together their function we could approximate their total function. However that’s not always the case because these genes might interact with one another—directly through binding or indirectly through competition—to perform their total function. These interactions might cause some deviation from our <strong>additive</strong> model of function; in genetics this is called <strong>epistasis</strong>.</p>
<p>This dataset is from <a href="">Prigge <em>et al</em>. 2020</a>, who collected measurements of several phenotypes for combinations of mutants in the TIR1/AFB auxin receptor genes which my group studies and engineers.</p>
</div>
<div id="read-in-data" class="section level3 hasAnchor" number="5.8.2">
<h3><span class="header-section-number">5.8.2</span> Read in data<a href="summarizing-data-examples.html#read-in-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="summarizing-data-examples.html#cb351-1" tabindex="-1"></a>root_growth_inh_0 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;data/elife-54740-fig1-figsupp3-data1-v2_data-only_tir1afb1.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;Primary Root Growth Inh EtOH&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span>
<span id="cb351-2"><a href="summarizing-data-examples.html#cb351-2" tabindex="-1"></a>root_growth_inh_20 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;data/elife-54740-fig1-figsupp3-data1-v2_data-only_tir1afb1.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;Primary Root Growth Inh 20 nM&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span>
<span id="cb351-3"><a href="summarizing-data-examples.html#cb351-3" tabindex="-1"></a>root_growth_inh_100 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;data/elife-54740-fig1-figsupp3-data1-v2_data-only_tir1afb1.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;Primary Root Growth Inh 100 nM&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span>
<span id="cb351-4"><a href="summarizing-data-examples.html#cb351-4" tabindex="-1"></a>root_growth_inh_500 <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;data/elife-54740-fig1-figsupp3-data1-v2_data-only_tir1afb1.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;Primary Root Growth Inh 500 nM&quot;</span>, <span class="at">skip =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="write-a-function-for-tidying-this-data" class="section level3 hasAnchor" number="5.8.3">
<h3><span class="header-section-number">5.8.3</span> Write a function for tidying this data<a href="summarizing-data-examples.html#write-a-function-for-tidying-this-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="summarizing-data-examples.html#cb352-1" tabindex="-1"></a>root_growth_plonger <span class="ot">&lt;-</span> <span class="cf">function</span>(data){</span>
<span id="cb352-2"><a href="summarizing-data-examples.html#cb352-2" tabindex="-1"></a>  data <span class="sc">|&gt;</span> </span>
<span id="cb352-3"><a href="summarizing-data-examples.html#cb352-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>line, </span>
<span id="cb352-4"><a href="summarizing-data-examples.html#cb352-4" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;genotype&quot;</span>, </span>
<span id="cb352-5"><a href="summarizing-data-examples.html#cb352-5" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;root_growth_mm&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb352-6"><a href="summarizing-data-examples.html#cb352-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="fu">str_extract</span>(genotype, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">s(.*)$&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb352-7"><a href="summarizing-data-examples.html#cb352-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="fu">str_remove_all</span>(batch, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">s&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb352-8"><a href="summarizing-data-examples.html#cb352-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genotype =</span> <span class="fu">str_remove</span>(genotype, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">s.*&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb352-9"><a href="summarizing-data-examples.html#cb352-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(batch), <span class="st">&quot;a&quot;</span>, batch)) <span class="sc">%&gt;%</span></span>
<span id="cb352-10"><a href="summarizing-data-examples.html#cb352-10" tabindex="-1"></a>  <span class="fu">na.omit</span>()</span>
<span id="cb352-11"><a href="summarizing-data-examples.html#cb352-11" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="tidy-the-individual-datasets" class="section level3 hasAnchor" number="5.8.4">
<h3><span class="header-section-number">5.8.4</span> Tidy the individual datasets<a href="summarizing-data-examples.html#tidy-the-individual-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="summarizing-data-examples.html#cb353-1" tabindex="-1"></a>root_growth_inh_0 <span class="ot">&lt;-</span> <span class="fu">root_growth_plonger</span>(root_growth_inh_0)</span>
<span id="cb353-2"><a href="summarizing-data-examples.html#cb353-2" tabindex="-1"></a>root_growth_inh_0<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb353-3"><a href="summarizing-data-examples.html#cb353-3" tabindex="-1"></a></span>
<span id="cb353-4"><a href="summarizing-data-examples.html#cb353-4" tabindex="-1"></a>root_growth_inh_20 <span class="ot">&lt;-</span> <span class="fu">root_growth_plonger</span>(root_growth_inh_20)</span>
<span id="cb353-5"><a href="summarizing-data-examples.html#cb353-5" tabindex="-1"></a>root_growth_inh_20<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="dv">20</span></span>
<span id="cb353-6"><a href="summarizing-data-examples.html#cb353-6" tabindex="-1"></a></span>
<span id="cb353-7"><a href="summarizing-data-examples.html#cb353-7" tabindex="-1"></a>root_growth_inh_100 <span class="ot">&lt;-</span> <span class="fu">root_growth_plonger</span>(root_growth_inh_100)</span>
<span id="cb353-8"><a href="summarizing-data-examples.html#cb353-8" tabindex="-1"></a>root_growth_inh_100<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="dv">100</span></span>
<span id="cb353-9"><a href="summarizing-data-examples.html#cb353-9" tabindex="-1"></a></span>
<span id="cb353-10"><a href="summarizing-data-examples.html#cb353-10" tabindex="-1"></a>root_growth_inh_500 <span class="ot">&lt;-</span> <span class="fu">root_growth_plonger</span>(root_growth_inh_500)</span>
<span id="cb353-11"><a href="summarizing-data-examples.html#cb353-11" tabindex="-1"></a>root_growth_inh_500<span class="sc">$</span>treatment <span class="ot">&lt;-</span> <span class="dv">500</span></span>
<span id="cb353-12"><a href="summarizing-data-examples.html#cb353-12" tabindex="-1"></a></span>
<span id="cb353-13"><a href="summarizing-data-examples.html#cb353-13" tabindex="-1"></a>root_growth_inh <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(root_growth_inh_0, </span>
<span id="cb353-14"><a href="summarizing-data-examples.html#cb353-14" tabindex="-1"></a>                             root_growth_inh_20, </span>
<span id="cb353-15"><a href="summarizing-data-examples.html#cb353-15" tabindex="-1"></a>                             root_growth_inh_100, </span>
<span id="cb353-16"><a href="summarizing-data-examples.html#cb353-16" tabindex="-1"></a>                             root_growth_inh_500)</span>
<span id="cb353-17"><a href="summarizing-data-examples.html#cb353-17" tabindex="-1"></a></span>
<span id="cb353-18"><a href="summarizing-data-examples.html#cb353-18" tabindex="-1"></a>root_growth_inh <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(batch) <span class="sc">%&gt;%</span> </span>
<span id="cb353-19"><a href="summarizing-data-examples.html#cb353-19" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">percent_inh =</span> root_growth_mm <span class="sc">/</span> </span>
<span id="cb353-20"><a href="summarizing-data-examples.html#cb353-20" tabindex="-1"></a>           <span class="fu">mean</span>(root_growth_mm[genotype <span class="sc">==</span> <span class="st">&quot;Col-0&quot;</span> <span class="sc">&amp;</span> treatment <span class="sc">==</span> <span class="dv">0</span>]) <span class="sc">*</span> <span class="dv">100</span>) <span class="ot">-&gt;</span></span>
<span id="cb353-21"><a href="summarizing-data-examples.html#cb353-21" tabindex="-1"></a>  root_growth_inh</span></code></pre></div>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="summarizing-data-examples.html#cb354-1" tabindex="-1"></a>root_growth_inh_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> genotype<span class="sc">*</span>treatment <span class="sc">+</span> batch, <span class="at">data =</span> root_growth_inh)</span>
<span id="cb354-2"><a href="summarizing-data-examples.html#cb354-2" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh_aov)</span></code></pre></div>
<pre><code>##                     Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## genotype             4   4807    1202   1.865  0.11536    
## treatment            1 423486  423486 657.249  &lt; 2e-16 ***
## batch                2   7356    3678   5.708  0.00355 ** 
## genotype:treatment   4  31237    7809  12.120 2.16e-09 ***
## Residuals          479 308635     644                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="summarizing-data-examples.html#cb356-1" tabindex="-1"></a>root_growth_inh<span class="sc">$</span>fac_treat <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(root_growth_inh<span class="sc">$</span>treatment)</span>
<span id="cb356-2"><a href="summarizing-data-examples.html#cb356-2" tabindex="-1"></a>root_growth_inh_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> genotype<span class="sc">*</span>fac_treat <span class="sc">+</span> batch, <span class="at">data =</span> root_growth_inh)</span>
<span id="cb356-3"><a href="summarizing-data-examples.html#cb356-3" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh_aov)</span></code></pre></div>
<pre><code>##                     Df Sum Sq Mean Sq  F value   Pr(&gt;F)    
## genotype             4   4807    1202   10.951 1.69e-08 ***
## fac_treat            3 708852  236284 2153.098  &lt; 2e-16 ***
## batch                2   1084     542    4.939  0.00754 ** 
## genotype:fac_treat   9   8981     998    9.093 9.95e-13 ***
## Residuals          472  51798     110                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb358-1"><a href="summarizing-data-examples.html#cb358-1" tabindex="-1"></a>root_growth_inh_aov.HSD <span class="ot">&lt;-</span> broom<span class="sc">::</span><span class="fu">tidy</span>(<span class="fu">TukeyHSD</span>(root_growth_inh_aov))</span>
<span id="cb358-2"><a href="summarizing-data-examples.html#cb358-2" tabindex="-1"></a>root_growth_inh_aov.HSD[<span class="fu">which</span>(root_growth_inh_aov.HSD<span class="sc">$</span>adj.p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span></code></pre></div>
<pre><code>## # A tibble: 121 × 7
##    term              contrast null.value estimate conf.low conf.high adj.p.value
##    &lt;chr&gt;             &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;     &lt;dbl&gt;       &lt;dbl&gt;
##  1 genotype          tir1-1-…          0     6.55     3.47      9.62    1.04e- 7
##  2 genotype          tir1-10…          0     7.57     3.42     11.7     7.93e- 6
##  3 genotype          tir1afb…          0     6.41     1.11     11.7     8.76e- 3
##  4 fac_treat         20-0              0   -36.5    -40.2     -32.9     2.81e-11
##  5 fac_treat         100-0             0   -75.8    -78.9     -72.7     2.81e-11
##  6 fac_treat         500-0             0   -93.9    -97.4     -90.5     2.81e-11
##  7 fac_treat         100-20            0   -39.3    -42.9     -35.6     2.81e-11
##  8 fac_treat         500-20            0   -57.4    -61.4     -53.4     2.81e-11
##  9 fac_treat         500-100           0   -18.2    -21.6     -14.7     2.81e-11
## 10 genotype:fac_tre… tir1-10…          0    15.4      2.19     28.6     6.04e- 3
## # ℹ 111 more rows</code></pre>
<p>Looks like there are no strong batch effects here.</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb360-1"><a href="summarizing-data-examples.html#cb360-1" tabindex="-1"></a>root_growth_inh <span class="sc">%&gt;%</span> <span class="fu">filter</span>(genotype <span class="sc">!=</span> <span class="st">&quot;tir1-10&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb360-2"><a href="summarizing-data-examples.html#cb360-2" tabindex="-1"></a>  <span class="fu">filter</span>(treatment <span class="sc">%in%</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">100</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb360-3"><a href="summarizing-data-examples.html#cb360-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TIR1 =</span> <span class="sc">!</span><span class="fu">str_detect</span>(genotype, <span class="st">&quot;tir1&quot;</span>), </span>
<span id="cb360-4"><a href="summarizing-data-examples.html#cb360-4" tabindex="-1"></a>                            <span class="at">AFB1 =</span> <span class="sc">!</span><span class="fu">str_detect</span>(genotype, <span class="st">&quot;afb1&quot;</span>)) <span class="ot">-&gt;</span></span>
<span id="cb360-5"><a href="summarizing-data-examples.html#cb360-5" tabindex="-1"></a>  root_growth_inh</span></code></pre></div>
<p>For now let’s look at the whole dataset.</p>
<div class="sourceCode" id="cb361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb361-1"><a href="summarizing-data-examples.html#cb361-1" tabindex="-1"></a>root_growth_inh.int <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> TIR1<span class="sc">*</span>AFB1<span class="sc">*</span>treatment, <span class="at">data =</span> root_growth_inh)</span>
<span id="cb361-2"><a href="summarizing-data-examples.html#cb361-2" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh.int)</span></code></pre></div>
<pre><code>##                      Df Sum Sq Mean Sq  F value   Pr(&gt;F)    
## TIR1                  1   6758    6758   48.681 2.69e-11 ***
## AFB1                  1    470     470    3.386  0.06692 .  
## treatment             1 369212  369212 2659.748  &lt; 2e-16 ***
## TIR1:AFB1             1   3158    3158   22.747 3.14e-06 ***
## TIR1:treatment        1   4190    4190   30.184 9.66e-08 ***
## AFB1:treatment        1   1456    1456   10.485  0.00137 ** 
## TIR1:AFB1:treatment   1      7       7    0.047  0.82862    
## Residuals           250  34704     139                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="summarizing-data-examples.html#cb363-1" tabindex="-1"></a>root_growth_inh.int <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> TIR1<span class="sc">*</span>AFB1<span class="sc">*</span>fac_treat, <span class="at">data =</span> root_growth_inh)</span>
<span id="cb363-2"><a href="summarizing-data-examples.html#cb363-2" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh.int)</span></code></pre></div>
<pre><code>##                      Df Sum Sq Mean Sq  F value   Pr(&gt;F)    
## TIR1                  1   6758    6758   48.681 2.69e-11 ***
## AFB1                  1    470     470    3.386  0.06692 .  
## fac_treat             1 369212  369212 2659.748  &lt; 2e-16 ***
## TIR1:AFB1             1   3158    3158   22.747 3.14e-06 ***
## TIR1:fac_treat        1   4190    4190   30.184 9.66e-08 ***
## AFB1:fac_treat        1   1456    1456   10.485  0.00137 ** 
## TIR1:AFB1:fac_treat   1      7       7    0.047  0.82862    
## Residuals           250  34704     139                      
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="summarizing-data-examples.html#cb365-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> root_growth_inh, <span class="fu">aes</span>(<span class="at">x =</span> fac_treat, <span class="at">y =</span> percent_inh, <span class="at">color =</span> genotype)) <span class="sc">+</span> </span>
<span id="cb365-2"><a href="summarizing-data-examples.html#cb365-2" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_jitterdodge</span>(<span class="at">jitter.width =</span> <span class="fl">0.2</span>))</span></code></pre></div>
<p><img src="05-1_Examples_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Strong case for an interaction, but not dependent on treatment, interesting.</p>
<p>To make this easier to explain we can repeat this analysis after stratifying by treatment.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="summarizing-data-examples.html#cb366-1" tabindex="-1"></a>root_growth_inh0.int <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> TIR1<span class="sc">*</span>AFB1, <span class="at">data =</span> <span class="fu">filter</span>(root_growth_inh, treatment <span class="sc">==</span> <span class="dv">0</span>))</span>
<span id="cb366-2"><a href="summarizing-data-examples.html#cb366-2" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh0.int)</span></code></pre></div>
<pre><code>##              Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## TIR1          1     43    42.5    0.26 0.611331    
## AFB1          1   2030  2030.3   12.40 0.000598 ***
## TIR1:AFB1     1   1700  1700.0   10.38 0.001619 ** 
## Residuals   127  20800   163.8                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>This matches expectations mostly, I would have expected AFB1 to have less effect, but in this model the double mutant is also contributing to this effect size. The interaction here is still a strong effect. 100 nM should be more telling even.</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="summarizing-data-examples.html#cb368-1" tabindex="-1"></a>root_growth_inh100.int <span class="ot">&lt;-</span> <span class="fu">aov</span>(percent_inh <span class="sc">~</span> TIR1<span class="sc">*</span>AFB1, <span class="at">data =</span> <span class="fu">filter</span>(root_growth_inh, treatment <span class="sc">==</span> <span class="dv">100</span>))</span>
<span id="cb368-2"><a href="summarizing-data-examples.html#cb368-2" tabindex="-1"></a><span class="fu">summary</span>(root_growth_inh100.int)</span></code></pre></div>
<pre><code>##              Df Sum Sq Mean Sq F value   Pr(&gt;F)    
## TIR1          1   9623    9623  85.129 9.93e-16 ***
## AFB1          1     84      84   0.746  0.38945    
## TIR1:AFB1     1   1407    1407  12.445  0.00059 ***
## Residuals   123  13903     113                     
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Very interesting, TIR1 has a very strong effect as expected at this high auxin concentration, whereas AFB1 does not, but the interaction term is significant at both 0 nM and 100 nM IAA.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summarizing-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-visualization.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection",
    "scroll_highlight": true
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
